### PR TITLE
Create registration

### PR DESCRIPTION
**Pull request by hieptq-0855**

### CODE REVIEW

**Comment by hieptq-0855** [comment]
ready

**Comment by hieptq-0855** [comment]
ready

**Comment by hieptq-0855** [comment]
ready

File: app/Http/Controllers/AjaxController.php [hunk 1]
```diff
@@ -3,33 +3,42 @@
...
     {
         $this->studentRepository = $studentRepository;
         $this->registrationRepository = $registrationRepository;
+        $this->semesterRepository = $semesterRepository;
     }
 
     function getClassTable(Request $rq) {
         $id = $rq['value'];
-        $classes = Subject::find($id)->getClass()->get();
+        $classes = Subject::findOrFail($id)->getClass()->get();
```
Comment (sonnnd-1393) [inline app/Http/Controllers/AjaxController.php]: repository

File: app/Repositories/GenerationRepository.php [hunk 2]
```diff
@@ -30,4 +31,11 @@ public function safeDelete($id)
...
     }
-}
+
+    public function paginate($number)
+    {
+        $generations = Generation::paginate($number);
+        
+        return $generations;
+    }
+}
```
Comment (sonnnd-1393) [inline app/Repositories/GenerationRepository.php]: dòng trắng

**Comment by hieptq-0855** [comment]
ready

**Comment by hieptq-0855** [comment]
ready

File: app/Http/Controllers/AjaxController.php [hunk 3]
```diff
@@ -85,6 +94,7 @@ function getSchedule(Request $rq)
             echo null;
         }
     }
+
     function getGenerationTable(Request $rq)
     {
         $generations = Generation::select(['id', 'name', 'begin_year']);
```
Comment (sonnnd-1393) [inline app/Http/Controllers/AjaxController.php]: repository

**Comment by hieptq-0855** [comment]
ready

**Pull request merged**